monitoring/messagesigning/README.md

```diff
@@ -0,0 +1,25 @@
+There is an environment variable titled `MESSAGE_SIGNING`. It must be set to `"true"` within `run_locally.sh` and `run_locally_scdsc.sh`. This will ensure that outgoing requests are signed and incoming requests, as well as received responses, have their message signing headers validated.
+## Setting up the `well-known` mock server
+
+### Keys
+Keys are under the folder - `monitoring/messagesigning/keys`. This folder should have two keys. Your `uft_priv.pem` key named as `mock_priv.pem`, and your public key used for valdiation named `mock_pub.der`. It is essential these two keys exist in that directory before running any message signing analysis.
+
+### Run well known server
+
+``./run_well_known.sh``  
+Public key will now be accessible at: http://host.docker.internal:8077/mock/scd/.well-known/uas-traffic-management/mock_pub.der
+
+## Creating the message signing report
+1. Inside of `monitoring/uss_qualifier`, run `./run_locally.sh configurations.dev.faa.uft.local_message_signing`
+2. Consult the created report within `monitoring/mock_uss/report/report_mock_uss_scdsc_messagesigning.json`
+
+## Analyzing the report
+No issues within the report signify that all incoming requests and received responses were successfully validated. There are several kinds of issues that could be generated. These include:  
+1. Missing message signing headers. If responses or incoming requests do not have all required message signing headers, an issue detailing which request/response was missing the headers will be recorded.
+2. Invalid content-digest message signing headers. If the `sha-512` hash value within the `content-digest` header does not match the one that `mock_uss` generates, an issue detailing the difference will be recorded. This is most likely a difference in `json` strings in the request/response bodies that get hashed. Check how your USS is generating the hash, vs how `mock_uss` is. For request/response bodies that are not null, `mock_uss` will take the `json.dumps(payload)` value and hash that.
+3. Invalid signatures. This kind of issue means the signature within the `x-utm-message-signature` header was not valid. This could happen for several reasons, it is bessed to check the logs within the `mock_uss` container for more information. 
+
+### Other info
+The `mock_uss` container will sign it's responses. It's important to check that it's signed responses can be validated by your USS. It's also important to test that the outgoing signed requests can be validated by your USS. In addition to this happy path testing, you can test that your USS rejects invalid requests, by turning the environment variable `MESSAGE_SIGNING` to "false". This will disable the signing of outgoing requests, and you should be able to see your USS handles the missing message signing headers request accordingly. It is also possible to leave the `MESSAGE_SIGNING` var to true, but use a faulty private key to sign. USS's should reject these faulty requests with a 403 status code.
```

Comment:
I would recommend to move this section in a new subsection called `Usage` in the new `Message signing` section in `mock_uss/README.md`.